sciris/sc_parallel.py

```diff
@@ -64,63 +64,117 @@ def loadbalancer(maxload=None, index=None, interval=None, maxtime=None, label=No
+            self.iterdict = iterdict  # A dictionary of values being iterated (may be None if iterval is not None)
+            self.args     = args      # Arguments passed directly to the function
+            self.kwargs   = kwargs    # Keyword arguments passed directly to the function
+            self.maxload  = maxload   # Maximum CPU load (ignored if ncpus is not None in parallelize()
             return None
 
 
 
-def parallelize(func=None, iterarg=None, itervar=None, args=None, kwargs=None, ncpus=None, maxload=None):
+def parallelize(func=None, iterarg=None, iterkwargs=None, args=None, kwargs=None, ncpus=None, maxload=None):
     '''
-    Shortcut for parallelizing a function. iterarg can be an iterable or an integer;
-    if the latter, it will run the function that number of times.
+    Shortcut for parallelizing a function. Most simply, acts as an shortcut for using
+    multiprocessing.Pool() or Queue(). However, this function can also iterate over more
+    complex arguments.
     
-    The function can either handle a fixed number of CPUs or allocate dynamically based
+    Either or both of iterarg or iterkwargs can be used. iterarg can be an iterable or an integer;
+    if the latter, it will run the function that number of times (which may be useful for
+    running "embarrassingly parallel" simulations). iterkwargs is a dict of iterables; each
+    iterable must be the same length (and the same length of iterarg, if it exists), and each 
+    dict key will be used as a kwarg to the called function.
+    
+    This function can either use a fixed number of CPUs or allocate dynamically based
     on load. If ncpus is None and maxload is None, then it will use the number of CPUs
-    returned by multiprocessing; if ncpus is not None, it will use the specified number;
-    if maxload is not None, it will allocate dynamically.
+    returned by multiprocessing; if ncpus is not None, it will use the specified number of CPUs;
+    if ncpus is None and maxload is not None, it will allocate the number of CPUs dynamically.
     
-    Example 1:
+    Example 1 -- simple usage as a shortcut to multiprocessing.map():
         def f(x):
             return x*x
         
-        results = parallelize(func=f, iterarg=[1,2,3], ncpus=3)
+        results = sc.parallelize(f, [1,2,3])
+    
+    Example 2 -- simple usage for "embarrassingly parallel" processing:
+        def rnd(i):
+            import pylab as pl
+            return pl.rand()
+        
+        results = sc.parallelize(rnd, 10)
+    
+    Example 3 -- using multiple arguments:
+        def f(x,y):
+            return x*y
+        
+        results = sc.parallelize(func=f, iterarg=[(1,2),(2,3),(3,4)], ncpus=3)
+    
+    Example 4 -- using multiple keyword arguments:
+        def f(x,y):
+            return x*y
+        
+        results = sc.parallelize(func=f, iterkwargs={'x':[1,2,3], 'y':[2,3,4]}, ncpus=3)
```

Comment:
I'm always on the fence about whether it's better to have a dict of lists, or a list of dicts/NamedTuples...I do lean towards the latter though in cases where users might want to add or remove examples. Presumably each entry of all the arrays goes together, so doing something like removing a run would mean having to remove an item from all of those lists. I guess it depends on whether you envisage people mainly writing out the code, in which case having to duplicate the keys in the source file e.g. `{'x':1,'y':2},{'x':2,'y':3}` is annoying, vs working with the iterkwargs programmatically, in which case I think `{'x':1,'y':2},{'x':2,'y':3}` might be easier to work with